packages/core/src/model.ts

```diff
@@ -0,0 +1,35 @@
+import { assign } from './actions';
+import { AssignAction, Assigner, EventObject, PropertyAssigner } from './types';
+import { mapValues } from './utils';
+
+export function createModel<TContext, TEvent extends EventObject>(
+  initialState: TContext
+): {
+  context: TContext;
+  withUpdaters: <
+    T extends {
+      [key: string]:
+        | Assigner<TContext, TEvent>
```

Comment:
just a note - `TEvent` is a wide-open type, it has to accept all events accepted by the machine because we cant know where this is going to be used in the machine itself (and thus what actual events can be received here)

Reply:
Yep, that's the intention.